### DISCUSSION TITLE
setFocus  trigger focus on the last input

### DISCUSSION THREAD

**Discussion by guyeilon** [react-hook-form/react-hook-form#10167] [Q&A] [answered]
Hi, I'm using some OTP package [https://www.npmjs.com/package/mui-one-time-password-input](url) that using the same name for each field.
when trigger the setFocus - I get focus the last input -instead of the first. - any ideas?

[screencast-localhost_3000-2023.03.23-10_06_56.webm](https://user-images.githubusercontent.com/87779981/227141292-ea397a1c-abdf-48cf-90b1-48686591d41b.webm)

**Comment by leapful** [comment, marked as answer]
@guyeilon 

It works fine on my demo: https://codesandbox.io/s/react-hook-form-with-mui-one-time-password-input-forked-eb61t2?file=/src/index.js

Please provide your codesandbox url to re-produce your issue.

**Reply by guyeilon** [reply]
Hi @leapful tanks for the response , adding `   ref={(elem) => {
                ref({
                  ...elem,
                  focus: () => elem.querySelector("input")?.focus()
                });
              }}`

worked for me!
